### PR TITLE
feat(SMN): add smn message template resource

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add smn message template resource
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:
  
**Release note**:

```release-note
  add smn message template resource
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/smn/' TESTARGS='-run TestAccSmnMessageTemplate_basic' 
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/smn/ -v -run TestAccSmnMessageTemplate_basic -timeout 360m -parallel 4 
=== RUN   TestAccSmnMessageTemplate_basic 
=== PAUSE TestAccSmnMessageTemplate_basic
=== CONT  TestAccSmnMessageTemplate_basic
--- PASS: TestAccSmnMessageTemplate_basic (14.92s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/smn       14.974s
```

### CODE REVIEW

File: docs/resources/smn_message_template.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+```hcl
+variable "name" {}
+variable "protocol" {}
+variable "content" {}
+
+resource "huaweicloud_smn_message_template" "test"{
+  name     = var.name
+  protocol = var.protocol
+  content  = var.content
```
Comment (hmbig2) [inline docs/resources/smn_message_template.md]: pls show the usage of parameter placeholders explicitly

File: docs/resources/smn_message_template.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+* `id` - The resource ID.
+
+* `tag_names` - Indicates the variable list. The variable name will be quoted in braces ({}) in the template.
+  When you use a template to send messages, you can replace the variable with any content.
+
+## Import
+
+The SMN message template can be imported using the `id`, e.g.
+
+```
```
Comment (deer-hang) [inline docs/resources/smn_message_template.md]: lost bash

File: docs/resources/smn_message_template.md (hunk 1 above)
Comment (houpeng80) [reply docs/resources/smn_message_template.md]: done

File: docs/resources/smn_message_template.md (hunk 2 above)
Comment (houpeng80) [reply docs/resources/smn_message_template.md]: done

File: huaweicloud/provider.go [hunk 3]
```diff
@@ -1040,8 +1040,9 @@ func Provider() *schema.Provider {
 			"huaweicloud_networking_secgroup_v2":      ResourceNetworkingSecGroup(),
 			"huaweicloud_networking_secgroup_rule_v2": ResourceNetworkingSecGroupRule(),
 
-			"huaweicloud_smn_topic_v2":        smn.ResourceTopic(),
-			"huaweicloud_smn_subscription_v2": smn.ResourceSubscription(),
+			"huaweicloud_smn_topic_v2":         smn.ResourceTopic(),
+			"huaweicloud_smn_subscription_v2":  smn.ResourceSubscription(),
+			"huaweicloud_smn_message_template": smn.ResourceSmnMessageTemplate(),
```
Comment (ShiChangkuo) [inline huaweicloud/provider.go]: please define the resource at L928

Comment (houpeng80) [reply huaweicloud/provider.go]: done

**Pull request merged**